rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -235,16 +273,115 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
+            is_reader);
+      }
+    }
+  }
+
+  std::unordered_set<std::string>
+  get_other_static_peers(
+    const std::map<std::string, std::vector<uint8_t>> &map) {
+    std::unordered_set<std::string> result;
+
+    auto static_peers_entry = map.find("staticpeers");
+    if (static_peers_entry != map.end()) {
+      auto static_peers_string = std::string(static_peers_entry->second.begin(),
+                                             static_peers_entry->second.end());
+      std::string delimiter(",");
+      size_t start = 0;
+      size_t end = 0;
+      while ((end = static_peers_string.find(delimiter, start)) !=
+             std::string::npos) {
+        if ((end - start) == 0) {
+          // Ignore empty peer list entries
+          RCUTILS_LOG_WARN_NAMED(
+              "rmw_fastrtps_shared_cpp",
+              "Empty entry found in static peers list from new participant");
+        } else {
+          auto hostname = static_peers_string.substr(start, end - start);
+          result.insert(hostname);
+          auto peer_aliases = rmw_fastrtps_shared_cpp::utils::get_peer_aliases(hostname);
+          for (const auto &a : peer_aliases) {
+            result.insert(a);
+          }
+          RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp", "Got static peer: %s", hostname.c_str());
+        }
+        start = end + delimiter.length();
+      }
+      if (start < static_peers_string.length() - 1) {
+        auto peer = static_peers_string.substr(start);
+        result.insert(peer);
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp", "Got last static peer: %s", peer.c_str());
+      }
+    }
+
+    return result;
+  }
+
+  bool should_ignore_host(const std::string &hostname,
+                          const std::unordered_set<std::string> &other_static_peers) {
+    bool should_ignore = false;
+
+    if (RMW_AUTOMATIC_DISCOVERY_RANGE_OFF ==
+          discovery_options_.automatic_discovery_range) {
+      return true;
+    }
+    if (hostname != my_hostname_) {
+      if (RMW_AUTOMATIC_DISCOVERY_RANGE_LOCALHOST ==
+              discovery_options_.automatic_discovery_range ||
+          RMW_AUTOMATIC_DISCOVERY_RANGE_DEFAULT ==
+              discovery_options_.automatic_discovery_range) {
+
+        if (!is_static_peer(hostname, other_static_peers)) {
+          should_ignore = true;
+        }
+      }
+    }
+
+    return should_ignore;
+  }
+
+
+  bool is_static_peer(const std::string &hostname,
+                      const std::unordered_set<std::string> &other_static_peers) {
+
+    using namespace rmw_fastrtps_shared_cpp;
+    // Check if the host is a static peer on our list
+    auto aliases = utils::get_peer_aliases(hostname);
+    for (size_t ii = 0; ii < discovery_options_.static_peers_count; ++ii) {
+      if (hostname == discovery_options_.static_peers[ii].peer_address) {
+        RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp",
+                                "Matching host in our static peer list");
```

Comment:
I wonder if this log message is actually useful or not, because without any of the related information for context it doesn't tell you much. Maybe either remove it or add the hostname and peer address or something.

Reply:
Better error message in 5b29937